File: app/src/main/java/com/homa_inc/androidlabs/Utils/HTTPUtil.java
package com.homa_inc.androidlabs.Utils;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.util.Log;

import java.io.BufferedInputStream;
import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class HTTPUtil {
    private static String stream = "";

    public  HTTPUtil(){}

    public String GetHTTPDataHandler(String urlString){
        try{
            URL url = new URL(urlString);
            HttpURLConnection urlConnection = (HttpURLConnection)url.openConnection();

            if(urlConnection.getResponseCode() == HttpURLConnection.HTTP_OK){
                InputStream inputStream = new BufferedInputStream(urlConnection.getInputStream());
                BufferedReader r = new BufferedReader(new InputStreamReader(inputStream));
                StringBuilder sb = new StringBuilder();
                String line;
                while((line = r.readLine()) != null)
                    sb.append(line);
                stream = sb.toString();
                urlConnection.disconnect();
            }
        } catch (Exception ex){
            return  null;
        }
        return stream;
    }

    public Bitmap GetBitmapFromUrl(String url){
        Bitmap img = null;
        try {
            InputStream in = new java.net.URL(url).openStream();
            img = BitmapFactory.decodeStream(in);
        } catch (Exception e) {
            Log.e("checkTAG", e.getMessage());
            e.printStackTrace();
        }
        return img;
    }


}

File: app/src/main/java/com/homa_inc/androidlabs/Utils/HttpUtil.kt
package com.homa_inc.androidlabs.Utils

import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.net.ConnectivityManager
import java.io.BufferedInputStream
import java.io.BufferedReader
import java.io.InputStreamReader
import java.lang.StringBuilder
import java.net.HttpURLConnection
import java.net.URL

class HttpUtil {
    companion object {
        fun dataHandler(urlString: String?): String?{
            var resultStream: String? = null
            try {
                val url = URL(urlString)
                val httpConnection = url.openConnection() as HttpURLConnection
                if(httpConnection.responseCode == HttpURLConnection.HTTP_OK){
                    val inputStream = BufferedInputStream(httpConnection.inputStream)
                    val reader = BufferedReader(InputStreamReader(inputStream))
                    val result = StringBuilder()
                    var line: String? = ""
                    while(line != null){
                        line = reader.readLine()
                        result.append(line)
                    }
                    resultStream = result.toString()
                    httpConnection.disconnect()
                }
            } catch (ex :Exception){
                return null
            }
            return resultStream
        }

        fun bitmapFromUrl(urlString: String): Bitmap? {
            var img: Bitmap? = null
            try{
                val input = URL(urlString).openStream()
                img = BitmapFactory.decodeStream(input)
            } catch (ex: Exception){
                ex.printStackTrace()
            }
            return img
        }

        @Suppress("DEPRECATION")
        fun hasConnection(context: Context): Boolean{
            val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            var networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI)
            if(networkInfo != null && networkInfo.isConnected)
                return true
            networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE)
            if(networkInfo != null && networkInfo.isConnected)
                return true
            networkInfo = connectivityManager.activeNetworkInfo
            if(networkInfo != null && networkInfo.isConnected)
                return true
            return false
        }
    }
}